rdkcentral/Ripple
# feat: modify user grant

fix: incorrect device grant path and app grant path

## What

fix incorrect device grant path and app grant path

## Why

Device grant and user grant functionality is broken because locally stored device grant and user grant info is unable to retrieve

## How

Use path.join to combine path directory and file name instead of using format!

## Test

Check `/opt/persistent/rippleapp_grants`  no longer present in log but  `/opt/persistent/ripple/app_grants`
Check `/opt/persistent/rippledevice_grants`  no longer present in log but  `/opt/persistent/ripple/device_grants`


## Checklist

- [ ] I have added tests that prove the feature works or the fix is effective


feat: 
sync modified user grant to cloud

## What

Sync modified user grant to cloud

## Why

Locally stored user grant is not in sync with cloud stored user grant

## How

Store modified user grant to cloud service

## Test

1. Store or modify user grant for an app
2. Check log to make sure user grant is stored to cloud service
3. Stop Ripple
4. Delete app_grants from persistent storage
5. Restart Ripple
6. Make usergrants.app request to check stored user grant is returned in response.
7. Modify user grant and check user grant is synced with latest value

## Checklist

- [ ] I have added tests that prove the feature works or the fix is effective
